### ISSUE TITLE
test-sigbus fails on /tmp being mounted as ext3, as ext3 doesn't support fallocate()

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#5833]
### Submission type

  - [X] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

git master

### Used distribution

Debian sid

Running the test-suite on a Debian armel portebox fails:

```
 54/227 test-sigbus                             FAIL     0.04 s

--- command ---
SYSTEMD_LANGUAGE_FALLBACK_MAP='/home/biebl/build/systemd/src/locale/language-fallback-map' PATH='/home/biebl/build/systemd/build-deb:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games' SYSTEMD_KBD_MODEL_MAP='/home/biebl/build/systemd/src/locale/kbd-model-map' /home/biebl/build/systemd/build-deb/test-sigbus
--- stderr ---
Assertion 'fallocate(fd, 0, 0, page_size() * 8) >= 0' failed at ../src/test/test-sigbus.c:41, function main(). Aborting.
```

**Comment by mbiebl** [comment]
Happens on armhf as well.

**Comment by evverx** [comment]
@mbiebl , could you run
```
mesontest -C your-build-directory --wrapper strace -v test-sigbus |& grep fallocate
```
, please?

**Comment by mbiebl** [comment]
```
fallocate(3, 0, 0, 32768)               = -1 EOPNOTSUPP (Operation not supported)
writev(2, [{iov_base="Assertion 'fallocate(fd, 0, 0, p"..., iov_len=120}, {iov_base="\n", iov_len=1}], 2Assertion 'fallocate(fd, 0, 0, page_size() * 8) >= 0' failed at ../src/test/test-sigbus.c:41, function main(). Aborting.
```

**Comment by evverx** [comment]
What filesystem is containing `/tmp`?

**Comment by mbiebl** [comment]
```
$ findmnt /tmp/
TARGET SOURCE          FSTYPE OPTIONS
/tmp   /dev/sdb2[/tmp] ext3   rw,relatime,errors=remount-ro,data=ordered
```

**Comment by evverx** [comment]
I think we could use `posix_fallocate` instead of `fallocate`.

@mbiebl , I am not sure how the issue is related to `meson`. We should probably remove the `meson` label.

**Comment by mbiebl** [comment]
@evverx You're right. Seems the official Debian mips buildds differ from the mips porter box.  I've never seen this failure for our autotools based builds, like in
https://buildd.debian.org/status/fetch.php?pkg=systemd&arch=armhf&ver=232-22&stamp=1490743194&raw=0
```
PASS: test-sigbus
```
That's why I assumed it was related to meson, which I used on the mips porter box to do a test build.

Dropping the meson label, as it indeed fails with autotools as well on that particular porter box.

**Comment by poettering** [comment]
uh, ext3 appears to not support fallocate(), I figure due to the fact that it got obsoleted by ext4...

As a quick fix, I'd propose upgrading to tmpfs or ext4 for /tmp. I'd also be willing to accept a patch that adds fallocate_fallback() or so, which falls back to writing NUL bytes with write() if fallocate() fails...

**Comment by mbiebl** [comment]
Unfortunately I don't have control over those porterboxes / buildds, so can't switch the file system.
I guess the bug report should be retitled if this is caused by the file system and not the architecture?

**Comment by evverx** [comment]
The fix is waiting in #5946. Please take a look.
